resharper/resharper-unity/src/ProjectExtensions.cs

```diff
@@ -40,7 +40,7 @@ public static bool HasUnityReference([NotNull] this ISolution solution)
         public static bool IsUnityProject([CanBeNull] this IProject project)
         {
             // Only VSTU adds the Unity project flavour. Unity + Rider don't, so we have to look at references
-            return project != null && (project.HasFlavour<UnityProjectFlavor>() || ReferencesUnity(project));
+            return project != null && (project.HasFlavour<UnityProjectFlavor>() || project.IsValid() && ReferencesUnity(project));
```

Comment:
Why would `project` be invalid here?

Reply:
missing/incompatible msbuild?

Reply:
Oh, would we still have a project in that case?

Reply:
I don't know, I can't reproduce it for real